src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs

```diff
@@ -11999,5 +11999,161 @@ public static void F()
             await VerifyItemIsAbsentAsync(source, "goo");
             await VerifyItemIsAbsentAsync(source, "Bar");
         }
+
+        [Fact]
+        public async Task ParameterAvailableInMethodAttributeNameof()
+        {
+            var source = @"
+class C
+{
+    [Some(nameof(p$$))]
+    void M(int parameter) { }
+}
+";
+            await VerifyItemExistsAsync(MakeMarkup(source), "parameter");
+
+            await VerifyItemIsAbsentAsync(MakeMarkup(source, languageVersion: "10"), "parameter");
+        }
+
+        [Fact]
+        public async Task ParameterNotAvailableInMethodAttributeNameofWithNoArgument()
+        {
+            var source = @"
+class C
+{
+    [Some(nameof($$))]
+    void M(int parameter) { }
+}
+";
+            // Tracked by https://github.com/dotnet/roslyn/issues/60812
+            await VerifyItemIsAbsentAsync(MakeMarkup(source), "parameter");
+        }
+
+        [Fact]
+        public async Task ParameterAvailableInMethodParameterAttributeNameof()
+        {
+            var source = @"
+class C
+{
+    void M([Some(nameof(p$$))] int parameter) { }
+}
+";
+            await VerifyItemExistsAsync(MakeMarkup(source), "parameter");
+        }
+
+        [Fact]
+        public async Task ParameterAvailableInLocalFunctionAttributeNameof()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        [Some(nameof(p$$))]
+        void local(int parameter) { }
+    }
+}
+";
+            // Speculation within attributes on local functions is broken
+            // Tracked by https://github.com/dotnet/roslyn/issues/60801
+            await VerifyItemExistsAsync(MakeMarkup(source), "parameter", skipSpeculation: true);
```

Comment:
Brainstormed with Aleksey and it's a bit involved to fix (I have notes from what Aleksey suggested). I likely won't be able to look at it before June. Might ask Rikki if he has bandwidth, since likely affects other scenarios with attributes on local functions and lambdas.